src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -909,6 +909,18 @@ void CodeGen::genCodeForBinary(GenTreeOp* treeNode)
     regNumber op1reg = op1->isUsedFromReg() ? op1->GetRegNum() : REG_NA;
     regNumber op2reg = op2->isUsedFromReg() ? op2->GetRegNum() : REG_NA;
 
+    if (varTypeIsFloating(treeNode->TypeGet()))
```

Comment:
This just hijacks all the floating-point types and forwards to `inst_RV_RV_TT`, the emit helpers (such as `emitIns_SIMD_R_R_R`) deal with the difference between VEX and non-VEX for `dst` and `op1Reg`.

Reply:
Great - I think that's the right approach.